google-cloud-bigquery/lib/google/cloud/bigquery/model/confusion_matrix.rb

```diff
@@ -0,0 +1,51 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+module Google
+  module Cloud
+    module Bigquery
+      class Model
+        # Confusion matrix for multi-class classification models.
+        class ConfusionMatrix
+          # Confidence threshold used when computing the entries of the
+          # confusion matrix.
+          #
+          # @return [Float]
+          def confidence_threshold
+            @gapi.confidence_threshold
+          end
+
+          # One row per actual label.
+          #
+          # @return [Hash]
+          def rows
```

Comment:
I do not fully understand how this is to be used. I would prefer to convert these rows to Ruby Hash objects, like we do for queries. Are these rows using the structure or `feature_columns` or `label_columns`?

Reply:
Is this a case of wanting a real model you can examine?  The hash approach makes sense here, as the rows are keyed by the label name, but you're dealing with list of entry stats as the value.  I can see about getting some models made public that might be more illustrative of the different types.

Reply:
When I say "convert these rows to Ruby Hash objects, like we do for queries", I don't mean exposing the underlying raw data as being discussed in the other conversation. I mean properly converting values to the correct Ruby objects. For example, a value of the string `"Infinity"` for a FLOAT64 column would be converted to the Ruby object `Float::INFINITY`.

That said, if we don't keep these classes then we don't need to be performing this conversion.

Reply:
I think the most complex thing is clustering metrics, where the model is going to communicate information about the clusters in terms of centroid objects.  However, the expectation around the statistics objects is for using them to evaluate the model training numerically, so conversion makes sense..  e.g. was the accuracy better in this model than before?  how many items with this label landed in this bucket?

Reply:
This is now deserialized JSON, so there is no need to coerce this data further.